pkg/ee/default-application-catalog/application_catalog.go

```diff
@@ -98,8 +99,24 @@ func DeployDefaultApplicationCatalog(ctx context.Context, logger *logrus.Entry,
 func applicationDefinitionReconcilerFactory(appDef *appskubermaticv1.ApplicationDefinition) kkpreconciling.NamedApplicationDefinitionReconcilerFactory {
 	return func() (string, kkpreconciling.ApplicationDefinitionReconciler) {
 		return appDef.Name, func(a *appskubermaticv1.ApplicationDefinition) (*appskubermaticv1.ApplicationDefinition, error) {
-			a.Labels = appDef.Labels
-			a.Annotations = appDef.Annotations
+			// Labels and annotations specified in the ApplicationDefinition installed on the cluster are merged with the ones specified in the ApplicationDefinition
+			// that is generated from the default application catalog.
+			kubernetes.EnsureLabels(a, appDef.Labels)
+			kubernetes.EnsureAnnotations(a, appDef.Annotations)
```

Comment:
For reviewer:

Labels and annotations are both important IMO. KKP admins can configure them for various different purposes and if we go simply with the values from the default application catalog then they will always be overridden to empty.